apps/dashboard/app/transaction/page.tsx

```diff
@@ -1,17 +1,59 @@
 import { getServerSession } from "next-auth";
 import React from "react";
+import { redirect } from "next/navigation";
+
 import { authOptions } from "../api/auth/[...nextauth]/route";
-import { apollo } from "../../services/graphql";
-import Sidebar from "@/components/side-bar";
 
-export default async function page() {
+import TransactionDetails from "@/components/transaction-details/transaction-details";
+import ContentContainer from "@/components/content-container";
+import {
+  fetchFirstTransactions,
+  fetchPaginatedTransactions,
+} from "@/services/graphql/queries/get-transactions";
+import TransactionCard from "@/components/transaction-details/transaction-item";
+import PageNumber from "@/components/transaction-details/page-number";
+
+interface TransactionDetailsSearchParams {
+  cursor: string;
+  direction: "next" | "previous";
+}
+
+export default async function page({
+  searchParams,
+}: {
+  searchParams: TransactionDetailsSearchParams;
+}) {
+  const { cursor, direction } = searchParams;
   const session = await getServerSession(authOptions);
   const isAuthed = session?.sub ?? false;
   const token = session?.accessToken;
-  const client = apollo(token).getClient();
   if (!isAuthed) {
-    return <div>not logged in</div>;
+    redirect("/api/auth/signin");
```

Comment:
should be using a middleware here? 

here is an example from the admin-panel: https://github.com/GaloyMoney/admin-panel/blob/main/middleware.ts

Reply:
Considered adding it to another PR, but anyway, added it to this one.